diff -Nur cocotron/Foundation/NSBundle.m cocotron.f/Foundation/NSBundle.m
--- cocotron/Foundation/NSBundle.m	2015-07-27 15:40:24.000000000 +0000
+++ cocotron.f/Foundation/NSBundle.m	2015-07-28 12:18:21.295802110 +0000
@@ -48,7 +48,21 @@
 #include <unistd.h>
 #endif
 
+#if !defined(LINUX)
 #import <objc/dyld.h>
+#else
+int _NSGetExecutablePath(char *path, uint32_t *capacity) {
+    if(*capacity < MAXPATHLEN)
+        return MAXPATHLEN;
+
+    if((*capacity = readlink("/proc/self/exe", path, *capacity)) < 0) {
+        *capacity = 0;
+        return -1;
+    }
+
+    return 0;
+}
+#endif
 
 #if defined(GCC_RUNTIME_3) || defined(APPLE_RUNTIME_4)
 
